backport/backport.py

```diff
@@ -0,0 +1,288 @@
+    """
+    log.info("backport.content_hash_values.start", variable_ids=variable_ids)
+    t = time.time()
+    content_hash = engine.execute(q, variable_ids=variable_ids).first()[0]
+    assert content_hash is not None
+    log.info("backport.content_hash_values.end", hash=content_hash, t=time.time() - t)
+    return content_hash
+
+
+def _checksums_match(short_name: str, md5_config: str, md5_values: str) -> bool:
+    try:
+        walden_ds_config = WaldenCatalog().find_one(short_name=f"{short_name}_config")
+        walden_ds_values = WaldenCatalog().find_one(short_name=f"{short_name}_values")
+    except KeyError:
+        # datasets not found in catalog
+        return False
+
+    return (
+        walden_ds_config.origin_md5 == md5_config
+        and walden_ds_values.origin_md5 == md5_values
+    )
+
+
+def _create_short_name(
+    short_name: Optional[str], dataset_id: int, variable_id: int
+) -> str:
+    """Create sensible short name for dataset."""
+    if short_name:
+        validate_underscore(short_name, "short-name")
+        # prepend dataset id to short name
+        return f"{dataset_id}_{short_name}"
+    else:
+        if variable_id:
+            return f"{dataset_id}_{variable_id}"
+        else:
+            return str(dataset_id)
+
+
+@click.command()
+@click.option("--dataset-id", type=int)
+@click.option("--variable-id", type=int)
+@click.option(
+    "--short-name", type=str, help="Short name of a dataset, must be under_score"
+)
+@click.option(
+    "--force/--no-force",
+    default=False,
+    type=bool,
+    help="Force overwrite even if checksums match",
+)
+@click.option("--dry-run/--no-dry-run", default=False, type=bool)
+def backport(
+    dataset_id: int, variable_id: int, short_name: str, force: bool, dry_run: bool
+) -> None:
+    engine = get_engine()
+
+    # get data from database
+    log.info("backport.loading_dataset", dataset_id=dataset_id)
+    ds = GrapherDatasetModel.load_dataset(engine, dataset_id)
+    log.info("backport.loading_variable", variable_id=variable_id or "all")
+    if variable_id:
+        vars = [GrapherVariableModel.load_variable(engine, variable_id)]
+    else:
+        # load all variables from a dataset
+        vars = GrapherDatasetModel.load_variables_for_dataset(engine, dataset_id)
+    variable_ids = [v.id for v in vars]
+
+    short_name = _create_short_name(short_name, dataset_id, variable_id)
+
+    config = _load_config(ds, vars)
+
+    # get checksums of config and values
+    md5_config = checksum_str(config.json(sort_keys=True))
+    md5_values = _content_hash_values(engine, variable_ids)
+
+    # if checksums of data and config are identical, skip upload
+    if not force:
+        if _checksums_match(short_name, md5_config, md5_values):
+            log.info("backport.skip", short_name=short_name, reason="checksums match")
+            return
+
+    # upload config to walden
+    log.info("backport.upload_config")
+    _upload_config_to_walden(
```

Comment:
I started implementing this and then realized it might be more trouble than it's worth. If I added optional `--upload` then what could happen is that:
- you run `backport` which updates your walden index with md5 of the file, but don't upload anything
- you run `backport --upload` hoping it would upload files to walden, but we find that md5 of the file is identical to md5 from walden index and it skip the dataset

When you start thinking about it, you realize that if we don't have md5 stored in S3 next to the file, there's no easy way how to do this safely.

However! We have `--dry-run` that lets you run the backport without adding it to walden index. That worked well for my development.

Reply:
Ok cool, this will do then!